frontend/src/containers/Ledger.js

```diff
@@ -153,34 +153,41 @@ export class Ledger extends Component {
 
   }
 
+  loadCollectiveData() {
+    const { collective, params, fetchProfile } = this.props;
+    if (collective && collective.id) return Promise.resolve(collective);
+    return fetchProfile(params.slug).then(result => {
+      return result.groups[result.slug];
+    });
+  }
+
   componentWillMount() {
     const { 
-      collective,
       fetchUsers,
       fetchPendingExpenses,
       fetchTransactions,
-      loadData,
       params
     } = this.props;
 
-    switch (params.action) {
-      case 'approve':
-        this.approveExp(params.expenseid);
-        break;
-      case 'reject':
-        this.rejectExp(params.expenseid);
-        break;
-    }
-
-    if (loadData) { // useful when not server-side rendered
-      fetchProfile(collective.slug);
-    }
-    Promise.all([
-      fetchUsers(collective.slug),
-      fetchPendingExpenses(collective.slug),
-      fetchTransactions(collective.slug)
-      ])
-    .then(() => scrollToExpense());
+    this.loadCollectiveData()
+      .then((collective) => {
+        this.collective = collective;
```

Comment:
I don't understand reducers. How do they work? Should I put the logic of fetching the collective data if not already rendered by the server there?
Isn't it overkill for what we need to do here? "If the user goes back to the collective page": how often would that happen and when it does, how much do we care?